src/sqlfluff/core/rules/base.py

```diff
@@ -203,6 +207,30 @@ def __eq__(self, other):
             return False
         return True  # pragma: no cover TODO?
 
+    @classmethod
+    def delete(cls, anchor: BaseSegment) -> "LintFix":
+        """Delete supplied anchor segment."""
+        return cls("delete", anchor)
+
+    @classmethod
+    def replace(cls, anchor: BaseSegment, edit: Iterable[BaseSegment]) -> "LintFix":
+        """Replace supplied anchor segment with the edit segments."""
+        return cls("replace", anchor, edit)
+
+    @classmethod
+    def create_before(
+        cls, anchor: BaseSegment, edit: Iterable[BaseSegment]
+    ) -> "LintFix":
+        """Create edit segments before the supplied anchor segment."""
+        return cls("create_before", anchor, edit)
+
+    @classmethod
+    def create_after(
+        cls, anchor: BaseSegment, edit: Iterable[BaseSegment]
+    ) -> "LintFix":
+        """Create edit segments after the supplied anchor segment."""
+        return cls("create_after", anchor, edit)
```

Comment:
This is the main addition of the PR, quite happy with how simple it is and think it makes using this class in the rules a lot nicer.

N.B. "edit" is renamed to "replace" as that better describes the fix type and avoids naming collisions with the existing edit instance attribute.